api/internal/localizer/locloader.go

```diff
@@ -0,0 +1,136 @@
+	scope filesys.ConfirmedDir
+
+	// localize destination
+	newDir filesys.ConfirmedDir
+
+	// loader at LocLoader's current directory
+	ifc.Loader
+
+	// whether LocLoader and all its ancestors are the result of local references
+	local bool
+}
+
+// ValidateLocArgs validates the arguments to kustomize localize.
+// On success, ValidateLocArgs returns a LocLoader at targetArg and the localize destination directory; otherwise, an
+// error.
+func ValidateLocArgs(targetArg string, scopeArg string, newDirArg string, fSys filesys.FileSystem) (*LocLoader, filesys.ConfirmedDir, error) {
+	// check earlier to avoid cleanup
+	spec, err := git.NewRepoSpecFromURL(targetArg)
+	if err == nil && spec.Ref == "" {
+		return nil, "", errors.Errorf("localize remote root '%s' missing ref query string parameter", targetArg)
+	}
+
+	// for security, should enforce load restrictions
+	ldr, err := loader.NewLoader(loader.RestrictionRootOnly, targetArg, fSys)
+	if err != nil {
+		return nil, "", errors.WrapPrefixf(err, "unable to establish localize target '%s'", targetArg)
+	}
+
+	scope, err := validateScope(scopeArg, targetArg, ldr, fSys)
+	if err != nil {
+		_ = ldr.Cleanup()
+		return nil, "", errors.WrapPrefixf(err, "invalid localize scope '%s'", scopeArg)
+	}
+
+	newDir, err := validateNewDir(newDirArg, ldr, spec, fSys)
+	if err != nil {
+		_ = ldr.Cleanup()
+		return nil, "", errors.WrapPrefixf(err, "invalid localize destination '%s'", newDirArg)
+	}
+
+	return &LocLoader{
+		fSys:   fSys,
+		scope:  scope,
+		newDir: newDir,
+		Loader: ldr,
+		local:  scope != "",
+	}, newDir, nil
+}
+
+// Scope returns the localize scope
+func (ll *LocLoader) Scope() string {
+	return ll.scope.String()
+}
+
+// Load returns the contents of path if path is a valid localize file.
+// Otherwise, Load returns an error.
+func (ll *LocLoader) Load(path string) ([]byte, error) {
+	// checks in root, and thus in scope
+	content, err := ll.Loader.Load(path)
+	if err != nil {
+		return nil, errors.WrapPrefixf(err, "invalid file reference")
+	}
+	if filepath.IsAbs(path) {
+		return nil, errors.Errorf("absolute paths not yet supported in alpha: file path '%s' is absolute", path)
+	}
+
+	abs := filepath.Join(ll.Root(), path)
+	dir, f, err := ll.fSys.CleanedAbs(abs)
+	if err != nil {
+		// should never happen
+		log.Fatalf(errors.WrapPrefixf(err, "cannot clean validated file path '%s'", abs).Error())
+	}
+	// target cannot reference newDir, as this load would've failed prior to localize;
+	// not a problem if remote because then reference could only be in newDir if repo copy,
+	// which will be cleaned, is inside newDir
+	if ll.local && dir.HasPrefix(ll.newDir) {
+		return nil, errors.Errorf("file path '%s' references into localize destination", dir.Join(f))
+	}
+
+	return content, nil
+}
+
+// New returns a LocLoader at path if path is a valid localize root.
+// Otherwise, New returns an error.
+func (ll *LocLoader) New(path string) (ifc.Loader, error) {
```

Comment:
Why does this return an `ifc.Loader` instead of a `LocLoader`?

Reply:
I tried returning `LocLoader`, but in that case Go doesn't seem to recognize `LocLoader` as an implementation of `ifc.Loader` because the spec of `New` isn't `New(string) (ifc.Loader, error)`.

Reply:
Why does `LocLoader` need to be an implementation of `ifc.Loader`?

If you return `ifc.Loader` here, then you won't be able to use the LocLoader's methods on the value that is returned.

Reply:
> Why does `LocLoader` need to be an implementation of `ifc.Loader`? You could also rename this function.
I want `LocLoader` to implement `ifc.Loader` because I want to pass it to `kusttarget` to `Load` the kustomization file. 

> If you return `ifc.Loader` here, then you won't be able to use the LocLoader's methods on the value that is returned.
Interestingly, the only method specific to `LocLoader` is `Scope`, which I don't need for `LocLoader`s other than the initial one at the target. If this is bad practice, I can make `Scope` a public field or return `scope` (in addition to `newDir`) is the factory method.

Reply:
> I want LocLoader to implement ifc.Loader because I want to pass it to kusttarget to Load the kustomization file.

I'm still not following why it's necessary to use kusttarget at all... Are you trying to use `loadKustFile`? We can define our own version of that for `LocLoader` it's a very simple function.

Reply:
I'm also wondering, if we don't need `LocLoader` once we start recursing (which is what I assume this `New` function is for), then what is the main purpose of `LocLoader`? Is it just there for arg validation?

Reply:
Ok, sorry for all my back and forth.

> Interestingly, the only method specific to LocLoader is Scope, which I don't need for LocLoaders other than the initial one at the target. If this is bad practice, I can make Scope a public field or return scope (in addition to newDir) is the factory method.

Let's go with the latter (return scope in addition to newDir) in the factory method. I think that will be the simplest way to move this forward.

Reply:
> I'm still not following why it's necessary to use kusttarget at all... Are you trying to use `loadKustFile`? We can define our own version of that for `LocLoader`, it's a very simple function.

I want to use the `Load` on `kusttarget` to get a `kustomization`

Reply:
> I want to use the Load on kusttarget to get a kustomization

Got it

Reply:
> I'm also wondering, if we don't need `LocLoader` once we start recursing (which is what I assume this `New` function is for), and you're fine with just using an `ifc.Loader`, then what is the main purpose of `LocLoader`? Is it mostly there for arg validation?

Yeah, you're right. It was mostly there so that the upper layer could get `LocLoader`-specific information before the recursion happens. Won't need it now that we've moved that information out.